ivy/functional/backends/tensorflow/general.py

```diff
@@ -338,11 +337,11 @@ def indices_where(x):
 def shape(
     x: Union[tf.Tensor, tf.Variable],
     as_array: bool = False,
-) -> Union[tf.Tensor, tf.Variable, TensorShape]:
+) -> Union[ivy.Shape, ivy.Array]:
     if as_array:
-        return tf.shape(x)
+        return ivy.array(x.shape)
```

Comment:
actually we should have `ivy.array(tf.shape(x))` or `ivy.Array(tf.shape(x))` if `tf.shape` returns a native shape instance.